Swift/BRCrypto/BRCryptoSystem.swift

```diff
@@ -496,107 +497,151 @@ public final class System {
-                            .filter { $0.blockchainID == blockchainModel.id }
-
-                        let currencyModels = try! currencyResult
-                            // On success, always merge `defaultCurrencies`
-                            .map { $0.unionOf (defaults) { $0.id }}
-                            // On error, use `defaults`
-                            .recover { (error: BlockChainDB.QueryError) -> [BlockChainDB.Model.Currency] in
-                                return defaults
-                            }.get()
-
-                        var associations: [Currency : Network.Association] = [:]
-
-                        // Update associations
-                        currencyModels
-                            // TODO: Only needed if getCurrencies returns the wrong stuff.
-                            .filter { $0.blockchainID == blockchainModel.id }
-                            .filter { $0.verified }
-                            .forEach { (currencyModel: BlockChainDB.Model.Currency) in
-                                // Create the currency
-                                let currency = Currency (uids: currencyModel.id,
-                                                         name: currencyModel.name,
-                                                         code: currencyModel.code,
-                                                         type: currencyModel.type,
-                                                         issuer: currencyModel.address)
-
-                                // Create the base unit
-                                let baseUnit: Unit = currencyModel.demoninations.first { 0 == $0.decimals }
-                                    .map { currencyDenominationToBaseUnit(currency: currency, model: $0) }
-                                    ?? currencyToDefaultBaseUnit (currency: currency)
-
-                                // Create the other units
-                                var units: [Unit] = [baseUnit]
-                                units += currencyModel.demoninations.filter { 0 != $0.decimals }
-                                    .map { currencyDenominationToUnit (currency: currency, model: $0, base: baseUnit) }
-
-                                // Find the default unit
-                                let maximumDecimals = units.reduce (0) { max ($0, $1.decimals) }
-                                let defaultUnit = units.first { $0.decimals == maximumDecimals }!
-
-                                // Update associations
-                                associations[currency] = Network.Association (baseUnit: baseUnit,
-                                                                              defaultUnit: defaultUnit,
-                                                                              units: Set<Unit>(units))
+        // Query for blockchains on the system.queue - thus system.configure() returns instantly
```

Comment:
To summarize the chunk of code change here, are there any business logic changes that Java need to mirror (aside from the new event) or is it just for the distpatch group/semaphore stuff?

Reply:
Only threading stuff.

Reply:
Now includes 'supported' and `.discoveredNetworks` stuff.  And, if called multiple times will check for a pre-existing network before creating on.